canister/src/unstable_blocks.rs

```diff
@@ -218,20 +245,25 @@ mod test {
     #[test]
     fn empty() {
         let anchor = BlockBuilder::genesis().build();
-        let utxos = UtxoSet::new(Network::Mainnet);
-        let mut forest = UnstableBlocks::new(&utxos, 1, anchor);
+        let network = Network::Mainnet;
+        let utxos = UtxoSet::new(network);
+        let mut forest = UnstableBlocks::new(&utxos, 1, anchor, Some(network));
         assert_eq!(peek(&forest), None);
         assert_eq!(pop(&mut forest), None);
     }
 
     #[test]
-    fn single_chain() {
-        let block_0 = BlockBuilder::genesis().build();
-        let block_1 = BlockBuilder::with_prev_header(block_0.header()).build();
-        let block_2 = BlockBuilder::with_prev_header(block_1.header()).build();
-
-        let utxos = UtxoSet::new(Network::Mainnet);
-        let mut forest = UnstableBlocks::new(&utxos, 1, block_0.clone());
+    fn single_chain_same_difficulties() {
+        let block_0 = BlockBuilder::genesis().build().with_mock_difficulty(1);
+        let block_1 = BlockBuilder::with_prev_header(block_0.header())
+            .build()
+            .with_mock_difficulty(1);
+        let block_2 = BlockBuilder::with_prev_header(block_1.header())
+            .build()
+            .with_mock_difficulty(1);
+        let network = Network::Mainnet;
+        let utxos = UtxoSet::new(network);
+        let mut forest = UnstableBlocks::new(&utxos, 2, block_0.clone(), Some(network));
```

Comment:
I don't think you need to mock the difficulty here. Instead, you can set the network to be `Network::Regtest`, and the difficulty of all these blocks will be 1.

Reply:
Good point, it is updated.